lib/exceptions.rb

```diff
@@ -28,7 +28,12 @@ class IncompatibleHost < StandardError; end
     class MachineryError < StandardError; end
 
     class UnknownScope < MachineryError; end
-    class UnknownOs < MachineryError; end
+    class UnknownOs < MachineryError
+      def to_s
+        "Unable to determine the operating system. There does not seem to be an" \
```

Comment:
That is true and I thought about it but it would make the error message more confusing. The new standard is /etc/os-release which is used if it is there.
If it is not there we also look for the SUSE and RHEL-Version. Do you think this should be mentioned?